src/Uno.UI/UI/Xaml/Window.skia.cs

```diff
@@ -66,49 +66,103 @@ internal void QueueInvalidateRender()
 			}
 		}
 
-		private static bool _isMeasureQueued = false;
-		private static bool _isMeasuring = false;
+		private bool _isMeasureWaiting = false;
+		private bool _isArrangeWaiting = false;
+		private bool _isMeasuringOrArranging = false;
 
 		internal static void InvalidateMeasure()
 		{
-			Current.InnerInvalidateMeasure();
+			Current.ScheduleInvalidateMeasureOrArrange(invalidateMeasure: true);
 		}
 
-		internal void InnerInvalidateMeasure()
+		internal static void InvalidateArrange()
 		{
+			Current.ScheduleInvalidateMeasureOrArrange(invalidateMeasure: false);
+		}
+
+		internal void ScheduleInvalidateMeasureOrArrange(bool invalidateMeasure)
+		{
+			if (invalidateMeasure)
+			{
+				if (_isMeasureWaiting)
+				{
+					// A measure is already queued
+					return;
+				}
+
+				_isMeasureWaiting = true;
+
+				if (_isArrangeWaiting)
+				{
+					// Since an arrange is already queued, no need to
+					// schedule something on the dispatcher
+					return;
+				}
+			}
+			else
+			{
+				if (_isArrangeWaiting)
+				{
+					// An arrange is already queued
+					return;
+				}
+
+				_isArrangeWaiting = true;
+
+				if (_isMeasureWaiting)
+				{
+					// Since a measure is already queued, no need to
+					// schedule something on the dispatcher
+					return;
+				}
+			}
+
 			if (_rootVisual != null)
 			{
-				if (!_isMeasureQueued)
+				CoreDispatcher.Main.RunAsync(CoreDispatcherPriority.Normal, RunMeasureAndArrange);
+			}
+		}
+
+		private void RunMeasureAndArrange()
```

Comment:
This somehow looks like `UIElement.UpdateLayout` except we don't have the 250 tries here, are we able to merge them ?

Reply:
I agree it's doing a pretty similar job and another PR should address it.